File: 205. Isomorphic Strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char,char> ums, umt;
        int ls = s.size(), lt = t.size();
        if (ls != lt) return false;
        
        for (int i = 0; i < ls; i++) {
            if (ums.find(s[i]) == ums.end() && umt.find(t[i]) == umt.end()) {
                ums[s[i]] = t[i];
                umt[t[i]] = s[i];
            }
            if (ums[s[i]] != t[i] || umt[t[i]] != s[i])
                return false;
        }
        
        return true;
    }
};

File: python/205. Isomorphic Strings.py
class Solution(object):
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        n = len(s)
        for i in range(n):
            if s.index(s[i]) != t.index(t[i]):
                return False
        return True